src/framework/ui/internal/navigationcontroller.cpp

```diff
@@ -1198,6 +1199,42 @@ void NavigationController::doTriggerControl()
     activeCtrl->trigger();
 }
 
+void NavigationController::onConfirm()
+{
+    TRACEFUNC;
+    MYLOG() << "====";
+    INavigationSection* activeSec = activeSection();
+    if (!activeSec) {
+        return;
+    }
+
+    INavigation::EventPtr e = Event::make(Event::Confirm);
+    activeSec->onEvent(e);
+    if (e->accepted) {
```

Comment:
Actually, never mind about my previous comment here (now deleted). 

This method looks good to me as it is right now, but we could extend it, so that if neither the section, nor the panel, nor the control handles the event, we additionally call `doTriggerControl`, to preserve the old behaviour where the new behaviour is not applicable.

Reply:
Another thought: we might want to invert the order: instead of trying the section first, then the panel, then the control, we might send the event to the most specific thing first, namely the control, and then to the panel, and then to the section.